torch/_torch_docs.py

```diff
@@ -219,6 +219,12 @@ def merge_dicts(*dicts):
 See :meth:`~Tensor.index_add_` for function description.
 """)
 
+add_docstr(torch.index_fill, r"""
+index_fill(input, dim, index, value, out=None) -> Tensor
+
+See :meth:`~Tensor.index_fill_` for function description.
+""")
```

Comment:
It's not clear to me that we want to reference the in-place function. Usually, we put the docs in the free function (which didn't exist before this PR).

cc @mruberry what do you think?